### PR TITLE
Refactors male & female gender buttons into dropdown list

### LINKED ISSUE #470: Update Gender Labels in Settings Screen
Currently, the only two options for genders are _Male_ and _Female_. We should either be adding a third button for _Non-Binary_ or, something that could be much more dynamic, turn it into a dropdown menu in case we add more in the future (e.g. genderqueer, gender-fluid, **trans male, **trans female). 

**The last two should only be added if we think that, for analytical purposes, knowing their sex-assigned at birth will be relevant. At the moment, it is not so we don't need to have that.

image

### PR DESCRIPTION
**Pull request by kevin-Price**
Refactors male & female gender buttons into dropdown list

Fixes mui components from overriding page css

## IMPORTANT: Please do not create a Pull Request without creating an issue first.

**Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.**

**Do not change the following line aside from replacing YOUR-ISSUE-HERE with the issue this PR fixes** 

fixes #470 

### Please provide enough information so that others can review your pull request:
This fixes issue 470. Removes the Male and Female buttons and uses a muiSelect component instead. An extra gender "non-binary" has been added. 

The muiSelect component was using its own css instead of inheriting from the page (body) css. Therefore, I needed to override those css selectors in Layout.jsx. 

### Explain the **details** for making this change. What existing problem does the pull request solve?
We want to be more gender inclusive, and provide an easy way to extend to more genders in the future. The muiSelect component gives us that ability.

### Test plan (required)

Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.
Before: 
Screen Shot 2020-04-12 at 2 02 03 PM
After: 
Screen Shot 2020-04-12 at 2 01 41 PM
Screen Shot 2020-04-12 at 2 03 20 PM

### Final Checklist

- [x] Have you bumped the version in `package.json`?
    - Second decimal for major change, third decimal for minor change. This can go past 10 i.e. 1.0.9 !=> 1.1.0, 1.0.9 => 1.0.10
    - [Read here for more info](https://semver.org/)
- [x] Have you added any new tests necessary?
- [x] Is your PR rebased off the most current master?
- [x] Have you squashed all commits, or if you will merge will you squash all commits?
- [x] Did you use yarn, not npm?
- [x] Did you use Material-UI wherever possible?
- [x] Did you format according to Prettier?
- [x] Did you run all of your most recent changes locally to make sure everything is working?

### CODE REVIEW

File: client/src/components/Layout.jsx [hunk 1]
```diff
@@ -9,6 +9,21 @@ import NavBar from './NavBar';
 import Login from './Login';
 
 const theme = createMuiTheme({
+  overrides: {
+    MuiSelect: {
+      select: {
+        '&&': {
+          // overrides .MuiSelect-select.MuiSelect-select
```
Comment (SomeMoosery) [inline client/src/components/Layout.jsx:16]: remove comment

File: client/src/components/OnboardUser.jsx [hunk 2]
```diff
@@ -145,40 +145,23 @@ const OnboardUser = props => {
...
+                  name="gender"
+                  value={formState.gender}
+                  onChange={handleChange}
+                  displayEmpty
+                  style={formState.gender !== '' ? { backgroundColor: '#82a4f8', outline: 'none', color: 'wheat' } : {}}
                 >
-                  Female
-                </button>
-              </ButtonGroup>
+                  <MenuItem value="" disabled>
```
Comment (SomeMoosery) [inline client/src/components/OnboardUser.jsx:157]: Do you need to set `value` here at all?

Comment (kevin-Price) [reply client/src/components/OnboardUser.jsx:157]: This does need to be here in order for the "click here" text to show.

File: client/src/components/Layout.jsx (hunk 1 above)
Comment (kevin-Price) [reply client/src/components/Layout.jsx]: Ill make this change and resubmit!

**Pull request merged**